substrate/frame/contracts/src/wasm/runtime.rs

```diff
@@ -839,9 +839,15 @@ impl<'a, E: Ext + 'a> Runtime<'a, E> {
 				} else {
 					self.read_sandbox_memory_as(memory, deposit_ptr)?
 				};
+				let read_only = flags.contains(CallFlags::READ_ONLY);
```

Comment:
If you do that here you don't have to repeat the logic twice.

```suggestion
				let read_only = flags.contains(CallFlags::READ_ONLY) || self.ext.is_read_only();
```

Reply:
I understand, but here are my two arguments: 
1. If `ext.call()` in `chain_extension` is used, the developer will need to understand these requirements and apply this condition to the call, otherwise he will corrupt our stack.
2. This is a call that can be either readonly (staticcall) or a standard call. I think that it is the stack's responsibility to put the frame with the correct permissions.
WDYT?

Reply:
You are right. This is a good point. Let's keep it like this then.